frontend/server/src/Controllers/Course.php

```diff
@@ -3100,6 +3102,82 @@ function () use ($course) {
         ];
     }
 
+    /**
+     * @return array{entrypoint: string, smartyProperties: array{payload: StudentsProgressv2Payload, title: \OmegaUp\TranslationString}}
+     *
+     * @omegaup-request-param int $length
+     * @omegaup-request-param int $page
+     * @omegaup-request-param string $course
+     */
+    public static function getStudentsProgressForSmarty(
+        \OmegaUp\Request $r
+    ): array {
+        $r->ensureIdentity();
+
+        $r->ensureOptionalInt('page');
+        $page = is_null($r['page']) ?? 1;
```

Comment:
```suggestion
        $page = $r->ensureOptionalInt('page') ?? 1;
```

ya nunca deberíamos usar `$r['...']` para nada!